test/tests/functional/pbs_stf.py

```diff
@@ -0,0 +1,761 @@
+        self.server.delete(jid)
+
+        a = {'Resource_List.max_walltime': '10:00:00',
+             'Resource_List.min_walltime': '10:00:00'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.server.delete(jid)
+
+        a = {'Resource_List.max_walltime': '09:00:00',
+             'Resource_List.min_walltime': '00:11:00'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+    def test_t_5_1_2(self):
+        """
+        STF job's max_walltime relative to resources_max.walltime
+        setting on server.
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        a = {'resources_max.walltime': '15:00:00'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        a = {'Resource_List.max_walltime': '16:00:00',
+             'Resource_List.min_walltime': '00:20:00'}
+        j = Job(TEST_USER, attrs=a)
+
+        error_msg = 'Job violates queue and/or server resource limits'
+        try:
+            jid = self.server.submit(j)
+        except PbsSubmitError as e:
+            self.assertTrue(error_msg in e.msg[0])
+
+        a = {'Resource_List.max_walltime': '16:00:00',
+             'Resource_List.min_walltime': '16:00:00'}
+        j = Job(TEST_USER, attrs=a)
+
+        try:
+            jid = self.server.submit(j)
+        except PbsSubmitError as e:
+            self.assertTrue(error_msg in e.msg[0])
+
+        a = {'Resource_List.max_walltime': '15:00:00',
+             'Resource_List.min_walltime': '15:00:00'}
+        j = Job(TEST_USER, attrs=a)
+
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+    def test_t_5_2_1(self):
+        """
+        Setting resources_max.min_walltime on a queue.
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        a = {'resources_max.min_walltime': '10:00:00'}
+        try:
+            self.server.manager(MGR_CMD_SET, QUEUE, a, id="workq")
+        except PbsManagerError as e:
+            self.assertTrue('Resource limits can not be set for the resource'
+                            in e.msg[0])
+
+    def test_t_5_2_2(self):
+        """
+        Setting resources_max.min_walltime on the server.
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        a = {'resources_max.min_walltime': '10:00:00'}
+        try:
+            rc = self.server.manager(MGR_CMD_SET, SERVER, a)
+        except PbsManagerError as e:
+            self.assertTrue('Resource limits can not be set for the resource'
+                            in e.msg[0])
+
+    def test_t6(self):
```

Comment:
This test looks good. I would suggest adding another test where setting min_walltime greater than max_walltime and making sure max_walltime is not unset.

Reply:
I extended the test to set min_walltime > the existing max_walltime and setting them both at the same time.